airframe-sql/src/main/scala/wvlet/airframe/sql/analyzer/TypeResolver.scala

```diff
@@ -380,8 +380,7 @@ object TypeResolver extends LogSupport {
           a.copy(expr = resolved)
         }
       case SingleColumn(a: Attribute, qualifier, _) if a.resolved =>
-        // Optimizes the nested attributes, but preserves qualifier in the parent
-        a.setQualifierIfEmpty(qualifier)
+        a.withQualifier(qualifier)
```

Comment:
Need to keep the qualifier if `SingleColumn` has it.

Reply:
If so, setQualifierIfEmpty method becomes unnecessary (no other usage). Let's remove it